marquez_airflow/extractors/bigquery_extractor.py

```diff
@@ -72,16 +62,45 @@ def _bq_table_name(self, bq_table):
     # convenience method
     def extract_on_complete(self, task_instance) -> [StepMetadata]:
         log.debug(f"extract_on_complete({task_instance})")
-
-        source = self._source()
-        bigquery_job_id = self._get_bigquery_job_id(task_instance)
-
         inputs = None
         outputs = None
         context = {
             'sql': self.operator.sql,
-            'bigquery.job_id': bigquery_job_id
         }
+        try:
+            source = self._source()
+            bigquery_job_id = self._get_bigquery_job_id(task_instance)
+        except Exception as e:
+            log.error(f"Cannot retrieve job details from BigQuery.Client. {e}",
+                      exc_info=True)
+            try:
+                sql_meta = SqlParser.parse(self.operator.sql)
+                log.debug(f"bigquery sql parsed and obtained meta: {sql_meta}")
+
+                return [StepMetadata(
+                    name=get_job_name(task=self.operator),
+                    context={
+                        'sql': self.operator.sql,
+                        'bigquery.sql.parsed.inputs':
+                            json.dumps(sql_meta.in_tables),
+                        'bigquery.sql.parsed.outputs':
+                            json.dumps(sql_meta.out_tables)
+                    }
+                )]
+            except Exception as e:
+                log.error(f"Cannot parse sql query. {e}",
+                          exc_info=True)
+                context['bigquery.extractor.error'] = \
+                    f"{e}: {traceback.format_exc()}"
+
+                return [StepMetadata(
+                        name=get_job_name(task=self.operator),
+                        inputs=inputs,
+                        outputs=outputs,
+                        context=context
+                        )]
+
+        context['bigquery.job_id'] = bigquery_job_id
```

Comment:
Ah! this is added back here. In this case you don't want to return `[StepMetadata(...` in the exceptions above as you lose this info. They should just add keys to the context and let the return on line 151 do the job.